Pacifique78/MyDiary
# #169572238 remove redundant data

#### What does this PR do?
This PR removes redundant data
#### Description of Task to be completed
>- Use same helpers, middleware for both version 1 & 2
>- Use 204 HTTP status code for deletion rather than 200
>- Fix errors in documentation
>- Fix get all entries response when there is no entry found
#### How should this be manually tested?
>- clone the repo
>- run `npm install` and `npm run dev` to start the server
>- use either postman or any other testing tool to see if  all the endpoints are working correctly
>- run `npm test` to see if the tests are passing
>- Check if the Travis test are passing
#### What are the relevant pivotal tracker stories?
#169572238
#### Screenshots (if appropriate)
Documentation Image
![Screenshot (124)](https://user-images.githubusercontent.com/50268181/68225404-09ed5880-fff9-11e9-9487-6cf2940df04d.png)
